src/java/org/apache/cassandra/security/DefaultCryptoProvider.java

```diff
@@ -39,11 +39,20 @@ public void installProvider()
     {
         try
         {
-            AmazonCorrettoCryptoProvider.install();
+            Class.forName("com.amazon.corretto.crypto.provider.AmazonCorrettoCryptoProvider");
```

Comment:
what if we want something completely different? (like we load other things…)

Reply:
@michaelsembwever then you would have a completely different implementation of `ICryptoProvider` which you reference in `crypto_provider`.

I was thinking about abstracting this logic so the same pattern would be used, "Amazon provider" would just override few values, the class to load, how installation and check is going to look like, but otherwise it would be all same for all specific crypto providers.